File: src/harpy/ast/statements/local_variable_declaration.py
from ..expressions import Expression
from .variable_declaration import VariableDeclaration


class LocalVariableDeclaration(VariableDeclaration):
    """Represents a local variable declaration, e.g. `local a` or `local b := 1`."""

    def __init__(self, name: str, assign_expr: Expression | None):
        super().__init__(variable_type="local", name=name, assign_expr=assign_expr)

File: Harpy/AST/Statements/LocalVariableDeclaration.cs
using Harpy.AST.Expressions;
using Harpy.Lexer;

namespace Harpy.AST.Statements;

/// <summary>
///     Represents a local variable declaration, e.g. <c>local a</c> or <c>local b := 1</c>.
/// </summary>
public class LocalVariableDeclaration(HarbourSyntaxToken scope, HarbourSyntaxToken name, Expression? assignment)
    : VariableDeclaration(scope, name, assignment);
